### ISSUE TITLE
A parser for ONNX and creating test-cases for ONNX to ONNX-MLIR translation

### ISSUE THREAD

**Issue by gramalingam** [onnx/onnx-mlir#469]
It will be convenient if we could use the filecheck utility for unit-tests for the translation of ONNX to ONNX-MLIR. This will be easier to do if we have a text-based syntax for ONNX models and a parser to create an ONNX model from text.

I created a PR in the ONNX repo with a proposed syntax and parser for this purpose: please see: https://github.com/onnx/onnx/pull/3194

**Comment by tjingrant** [comment]
@gramalingam that makes a lot of sense!

One thing I had worked on (and hope to continue pushing forward with in the near future) is end-to-end tests, i.e., tests converting tensorflow/pytorch checkpoints to onnx models using converters and verify they can be parsed correctly by onnx-mlir. I believe these end-to-end conversion tests will complement the unit test suites mentioned here nicely.

Please let me and other collaborators know how best to support your effort; as a starter, I can help with upgrading the onnx submodule version once your patch is merged into the ONNX master branch.

**Comment by tungld** [comment]
This is a good idea. It could help solve this issue #218 also.
